### ISSUE TITLE
Custom Homebrew install location and "Cannot find libz"

### ISSUE THREAD

**Issue by inakiabt** [Homebrew/homebrew-php#3569]
## Description

I've been using homebrew since day 1 installed via [boxen](https://github.com/boxen). **It's installed in a custom location (not the default): /opt/boxen/homebrew**.
Since some months ago, I couldn't upgrade my php version (5.4) because of this "Cannot find libz" error. 
```bash
==> ./configure --prefix=/opt/boxen/homebrew/Cellar/php56/5.6.25_1 --localstatedir=/opt/boxen/homebrew/var --sysconfdir=/opt/boxen/homebrew/etc/php/5.6 --with-config-file-path=/opt/boxen/homebrew/etc/ph
Last 15 lines from /Users/inakiabt/Library/Logs/Homebrew/php56/01.configure:
checking whether to use system default cipher list instead of hardcoded value... no
checking for krb5-config... /usr/bin/krb5-config
checking for DSA_get_default_method in -lssl... no
checking for X509_free in -lcrypto... yes
checking for RAND_egd... no
checking for pkg-config... no
checking for OpenSSL version... >= 0.9.6
checking for CRYPTO_free in -lcrypto... yes
checking for SSL_CTX_set_ssl_version in -lssl... yes
checking for PCRE library to use... bundled
checking whether to enable the SQLite3 extension... yes
checking bundled sqlite3 library... yes
checking for ZLIB support... yes
checking if the location of ZLIB install directory is defined... no
configure: error: Cannot find libz
```

I've tried everything I found about related issues:
  - `xcode-select --install` (output: `xcode-select: error: command line tools are already installed, use "Software Update" to install updates`)
  - `brew link zlib --force`
  - force linking the lib `sudo ln -s /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include /usr`
  - reinstall zlib with `brew install --force`

Nothing worked.

Reading my compilation error logs I see this `--with-zlib=/usr`. Looking for the library there, I found:
ls -l /usr/lib/libz*
```
lrwxr-xr-x  1 root  wheel      12 Oct  1  2015 /usr/lib/libz.1.1.3.dylib -> libz.1.dylib
lrwxr-xr-x  1 root  wheel      12 Oct  1  2015 /usr/lib/libz.1.2.5.dylib -> libz.1.dylib
-rwxr-xr-x  1 root  wheel  177728 Mar 12 05:37 /usr/lib/libz.1.dylib
lrwxr-xr-x  1 root  wheel      12 Oct  1  2015 /usr/lib/libz.dylib -> libz.1.dylib
```
ls -l /usr/local/lib/libz*
```
ls: /usr/local/lib/libz*: No such file or directory
```
ls -l /opt/boxen/homebrew/include/zlib*
```
lrwxr-xr-x  1 inakiabt  staff  35 Sep  5 10:47 /opt/boxen/homebrew/include/zlib.h -> ../Cellar/zlib/1.2.8/include/zlib.h
```
ls -l /opt/boxen/homebrew/lib/libz*
```
lrwxr-xr-x  1 inakiabt  staff  31 Sep  5 10:47 /opt/boxen/homebrew/lib/libz.a -> ../Cellar/zlib/1.2.8/lib/libz.a
lrwxr-xr-x  1 inakiabt  staff  32 Sep  5 10:47 /opt/boxen/homebrew/lib/libz.so -> ../Cellar/zlib/1.2.8/lib/libz.so
lrwxr-xr-x  1 inakiabt  staff  34 Sep  5 10:47 /opt/boxen/homebrew/lib/libz.so.1 -> ../Cellar/zlib/1.2.8/lib/libz.so.1
lrwxr-xr-x  1 inakiabt  staff  38 Sep  5 10:47 /opt/boxen/homebrew/lib/libz.so.1.2.8 -> ../Cellar/zlib/1.2.8/lib/libz.so.1.2.8
```

I'm guessing it could be related to the non-default location install of my homebrew and the `--with-zlib` argument pointing to `/usr` where I'm seeing the library it's not completely there (I'm guessing that because of above `ls -l /usr/local/lib/libz*` result)
Is there a way to pass a custom location of `zlib` (`--with-zlib`) when installing PHP?
What else can I do?
I'm desperate.

## Context and Logs

Parameter | Value
--------- | ------------------
**OS X Version:**       | 10.11.6
**Homebrew Version:**   | 0.9.9 (git revision efd29e; last commit 2016-09-05)
**PHP Version in use:** | 5.5.36
**Xcode Version:**      | 7.3.1

**Output of gcc -v:**
```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/c++/4.2.1
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
**Output of php -v:**
```
PHP 5.4.42 (cli) (built: Jun 24 2015 21:00:34)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.4.0, Copyright (c) 1998-2014 Zend Technologies
```
**Build logs:** https://gist.github.com/inakiabt/e85324759a697e0fa0974c30881567fe

**Comment by javian** [comment]
Could this help ? https://github.com/Homebrew/legacy-homebrew/issues/50334
The libz headers comes from the developer tools so the key is getting that right.

**Comment by inakiabt** [comment]
@javian It helped and now is working! Thank you so much.

I never liked that `CLT: N/A` in my `brew config` (output)[https://gist.github.com/inakiabt/e85324759a697e0fa0974c30881567fe#file-00-config-out-L22], but everything was working so...

Looking into that issue I found that `pkgutil --pkgs | grep CLT` must show my Command Line Tools version: Empty output in my case. But `xcode-select --install` was telling me that it's installed!
So, I started looking how to reinstall it and finished downloading it manually from https://developer.apple.com/download/more/. After install it, `brew upgrade php54` worked like a charm 💯
